### PR TITLE
Enable Strict Null Checks

### PR DESCRIPTION
**Pull request by casey-chow**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] API docs have been updated (by executing `npm run documentation`)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Other... Please describe:

## What is the current behavior?

N/A.

## What is the new behavior?

N/A.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

It does lead to API changes, but since it only expands the types being used, there shouldn't be anything API-incompatible.

## Other information

This PR enables strict null checks and modifies interfaces to pass type checking. In general, I just added `undefined` to properties that could be `undefined` in runtime.

### CODE REVIEW

File: src/Configuration.ts [hunk 1]
```diff
@@ -159,11 +161,11 @@ export class Configuration implements IConfiguration {
     this.partialUpdate({ url });
   }
 
-  public get project(): string {
+  public get project(): string | undefined {
```
Comment (rijkvanzanten) [inline src/Configuration.ts]: Project should always be set

File: src/Configuration.ts [hunk 2]
```diff
@@ -185,11 +187,11 @@ export class Configuration implements IConfiguration {
     this.internalConfiguration.persist = persist;
   }
 
-  public get mode(): AuthModes {
+  public get mode(): AuthModes | undefined {
```
Comment (rijkvanzanten) [inline src/Configuration.ts]: Auth mode should also always have a value (`cookie` vs `jwt`)

**Comment by adamkobor** [comment]
Hey @rijkvanzanten, 
any update on this?
(Thanks for your great work :) )

**Comment by rijkvanzanten** [comment]
No updates as of now. There were two fields that looked wrong to me, but those were never changed. Wanna lift this to a new PR and finish it up @adamkobor? 🙂

**Comment by adamkobor** [comment]
@rijkvanzanten sure, I also did a lot more :) 
https://github.com/directus/sdk-js/pull/145

**Comment by rijkvanzanten** [comment]
Closed in favor of https://github.com/directus/sdk-js/pull/145

**Pull request closed without merging**